plugins/pulp_rpm/plugins/importers/yum/config_validate.py

```diff
@@ -25,3 +46,69 @@ def validate(config):
             msg += failure_message + '\n'
         msg = msg.rstrip()  # remove the trailing \n
         return False, msg
+
+
+def _validate_allow_keys(repo, config, error_messages):
+    """
+    Validates that passed signature keys have proper lenth
+
+    :param repo: repository to validate the config for
+    :type  repo: pulp.plugins.model.Repository
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :param error_messages: errors encountered during validation
+    :error_messages: list
+    """
+
+    allow_keys = config.get('allow_keys', set())
+    for key in allow_keys:
+        if len(key) != 8:
+            msg = _('Signature key %s should be 8 characters long') % key
+            error_messages.append(msg)
+
+
+def _validate_lazy_compatibility(repo, config, error_messages):
+    """
+    Validates lazy sync and signature policy compatibility.
+    To extract the signature from the package, we need this package been downloaded.
+
+    :param repo: repository to validate the config for
+    :type  repo: pulp.plugins.model.Repository
+
+    :param config: configuration instance passed to the importer
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+
+    :param error_messages: errors encountered during validation
+    :error_messages: list
+    """
+
+    download_policy = config.get('download_policy', 'immediate')
+    allow_unsigned = config.get('allow_unsigned', True)
```

Comment:
i agree that require_signature is better naming, the rest that you pointed out is already working the way you described